korma/src/commonMain/kotlin/com/soywiz/korma/geom/Vector3D.kt

```diff
@@ -5,95 +5,190 @@ import com.soywiz.korma.interpolation.interpolate
+        (l.y * r.z - l.z * r.y),
+        (l.z * r.x - l.x * r.z),
+        (l.x * r.y - l.y * r.x)
     )
-    override fun equals(other: Any?): Boolean = (other is Vector3D) && almostEquals(this.x, other.x) && almostEquals(this.y, other.y) && almostEquals(this.z, other.z) && almostEquals(this.w, other.w)
+
+    // region data class / JVM
+
+    override fun equals(other: Any?): Boolean = (other is Vector3D) && almostEquals(this.x, other.x) && almostEquals(this.y, other.y) && almostEquals(this.z, other.z)
     override fun hashCode(): Int = data.contentHashCode()
+    override fun toString(): String = "(${x.niceStr}, ${y.niceStr}, ${z.niceStr})"
+
+    // endregion
+
+    // region Operator overloading
+
+    /**
+     * Get data value by index (only 0, 1, and 2 are valid indices)
+     */
+    operator fun get(index: Int): Float = data[index]
+
+    /**
+     * Set data value by index (only 0, 1, and 2 are valid indices)
+     */
+    operator fun set(index: Int, value: Float) = run { data[index] = value }
+
+    operator fun minus(other: Vector3D): Vector3D = Vector3D(x - other.x, y - other.y, z - other.z)
+
+    operator fun minusAssign(other: Vector3D) {
+        x -= other.x
+        y -= other.y
+        z -= other.z
+    }
+
+    operator fun plus(other: Vector3D): Vector3D = Vector3D(x + other.x, y + other.y, z + other.z)
+
+    operator fun plusAssign(other: Vector3D) {
+        x += other.x
+        y += other.y
+        z += other.z
+    }
+
+    /**
+     * Dot product.
+     */
+    operator fun times(other: Float): Vector3D = Vector3D(x * other, y * other, z * other)
+
+    /**
+     * Dot product.
+     */
+    operator fun timesAssign(other: Float) {
+        x *= other
+        y *= other
+        z *= other
+    }
+
+    /**
+     * Cross product
+     */
+    operator fun times(other: Vector3D): Vector3D = Vector3D(
+        (y * other.z - z * other.y),
+        (z * other.x - x * other.z),
+        (x * other.y - y * other.x)
+    )
+
+    /**
+     * Cross product
+     */
+    operator fun timesAssign(other: Vector3D) {
+        x = (y * other.z - z * other.y)
+        y = (z * other.x - x * other.z)
+        z = (x * other.y - y * other.x)
+    }
+
+    /**
+     * Inverse dot product.
+     */
+    operator fun div(other: Float): Vector3D = Vector3D(x / other, y / other, z / other)
+
+    /**
+     * Inverse dot product.
+     */
+    operator fun divAssign(other: Float) {
+        x /= other
+        y /= other
+        z /= other
+    }
 
-    override fun toString(): String = if (w == 1f) "(${x.niceStr}, ${y.niceStr}, ${z.niceStr})" else "(${x.niceStr}, ${y.niceStr}, ${z.niceStr}, ${w.niceStr})"
+    // endregion
 }
 
 inline class IntVector3(val v: Vector3D) {
     val x: Int get() = v.x.toInt()
     val y: Int get() = v.y.toInt()
     val z: Int get() = v.z.toInt()
-    val w: Int get() = v.w.toInt()
 }
 
-fun Vector3D.asIntVector3D() = IntVector3(this)
+fun Vector3D.toIntvector() = IntVector3(this)
```

Comment:
Missing case on the `v`
`.toIntvector`
->
`.toIntVector`

Also, the `as` means something. Let me explain the rationale:
* `to` was used when performing a conversion, potentially costly
* `as` was used when inline-wrapping another class, to state that there is like a reinterpretation with no conversion cost